.github/workflows/rust.yml

```diff
@@ -99,6 +99,14 @@ jobs:
           rust-version: stable
       - name: Run tests (excluding doctests)
         run: cargo test --lib --tests --bins --features avro,json,backtrace
+        env:
+          # do not produce debug symbols to keep memory usage down
+          # hardcoding other profile params to avoid profile override values
+          # More on Cargo profiles https://doc.rust-lang.org/cargo/reference/profiles.html?profile-settings#profile-settings
+          RUSTFLAGS: "-C debuginfo=0 -C opt-level=0 -C incremental=false -C codegen-units=256"
```

Comment:
https://github.com/apache/arrow-datafusion/blob/1179a76567892b259c88f08243ee01f05c4c3d5c/.github/actions/setup-builder/action.yaml#L45-L50

I think some of these flags are already enabled, technically? Just for Linux though, not Mac/Windows

(Well debuginfo was set to 0, not sure on incremental)

Reply:
Thats good point, I would like to have configuration in a single place which is `rust.yml`, I'll create a followup to remove CARGO_INCREMENTAL and backtraces and other rust conf from `action.yaml`